documentation/dataplane/dash-flow/DASH_FLOW_API_HLD.md

```diff
@@ -0,0 +1,500 @@
+| ----------------------------------------- | ------------------ | ------------------------------------------- |
+| SAI_FLOW_ENTRY_ATTR_IS_BIDIRECTIONAL_FLOW | `bool`             | Indicates if the flow is bidirectional      |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_IP_PROTOCOL   | `sai_uint8_t`      | IP protocol number for the reverse flow     |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_IP_ADDR       | `sai_ip_address_t` | Source IP address for the reverse flow      |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_IP_ADDR       | `sai_ip_address_t` | Destination IP address for the reverse flow |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_SRC_L4_PORT   | `sai_uint16_t`     | L4 source port for the reverse flow         |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_DST_L4_PORT   | `sai_uint16_t`     | L4 destination port for the reverse flow    |
+
+#### Flow encap
+
+These are the related attributes of flow encapsulation.
+
+| Attribute name                         | Type                       | Description                                     |
+| -------------------------------------- | -------------------------- | ----------------------------------------------- |
+| SAI_FLOW_ENTRY_ATTR_DEST_VNET_VNI      | `sai_uint32_t`             | Destination VNI                                 |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY_SIP       | `sai_uint32_t`             | Source IP address in the underlay network       |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY_DIP       | `sai_uint32_t`             | Destination IP address in the underlay network  |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY_SMAC      | `sai_mac_t`                | Source MAC address in the underlay network      |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY_DMAC      | `sai_mac_t`                | Destination MAC address in the underlay network |
+| SAI_FLOW_ENTRY_ATTR_DASH_ENCAPSULATION | `sai_dash_encapsulation_t` | Encapsulation method for DASH traffic           |
+
+#### Flow rewrite
+
+These are the related attributes of flow rewrite.
+
+| Attribute name               | Type               | Description                                                  |
+| ---------------------------- | ------------------ | ------------------------------------------------------------ |
+| SAI_FLOW_ENTRY_ATTR_IS_IPV6  | `bool`             | Indicates whether the flow is IPv6 (true) or IPv4 (false).   |
+| SAI_FLOW_ENTRY_ATTR_D_MAC    | `sai_mac_t`        | Destination MAC address for the flow entry.                  |
+| SAI_FLOW_ENTRY_ATTR_SIP      | `sai_ip_address_t` | Source IP address for the flow entry, supporting both IPv4 and IPv6. |
+| SAI_FLOW_ENTRY_ATTR_DIP      | `sai_ip_address_t` | Destination IP address for the flow entry, supporting both IPv4 and IPv6. |
+| SAI_FLOW_ENTRY_ATTR_SIP_MASK | `sai_ip_address_t` | Subnet mask for the source IP address.                       |
+| SAI_FLOW_ENTRY_ATTR_DIP_MASK | `sai_ip_address_t` | Subnet mask for the destination IP address.                  |
+
+### Flow Bulk Get Session
```

Comment:
need to add event notification for session state change.

Reply:
can refer to this state here: https://github.com/r12f/SONiC/blob/user/r12f/ha2/doc/smart-switch/high-availability/smart-switch-ha-detailed-design.md#2333-flow-sync-session-states